docs/proposals/federation.md

```diff
@@ -404,36 +486,99 @@ These are often left implicit by customers, but are worth calling out explicitly
 1. Software failure isolation between Kubernetes clusters should be
    retained as far as is practically possible.  The federation system
    should not materially increase the failure correlation across
-   clusters.  For this reason the federation system should ideally be
-   completely independent of the Kubernetes cluster control software,
-   and look just like any other Kubernetes API client, with no special
-   treatment.  If the federation system fails catastrophically, the
-   underlying Kubernetes clusters should remain independently usable.
+   clusters.  For this reason the federation control plane software
+   should ideally be completely independent of the Kubernetes cluster
+   control software, and look just like any other Kubernetes API
+   client, with no special treatment.  If the federation control plane
+   software fails catastrophically, the underlying Kubernetes clusters
+   should remain independently usable.
 1. Unified monitoring, alerting and auditing across federated Kubernetes clusters.
 1. Unified authentication, authorization and quota management across
    clusters (this is in direct conflict with failure isolation above,
    so there are some tough trade-offs to be made here).
 
-## Proposed High-Level Architecture
-
-TBD: All very hand-wavey still, but some initial thoughts to get the conversation going...
+## Proposed High-Level Architectures
+
+Two distinct potential architectural approaches have emerged from discussions
+thus far:
+
+1. An explicitly decoupled and hierarchical architecture, where the
+Federation Control Plane sits logically above a set of independent
+Kubernetes clusters, each of which is unaware of the other clusters,
+and of the Federation Control Plane itself (other than to the extent
+that it is an API client much like any other).  One possible example
+of this general architecture is illustrated below, and will be
+referred to as the "Decoupled, Hierarchical" approach.
+1. A more monolithic architecture, where a single instance of the
+Kubernetes control plane itself manages a single logicial cluster
+composed of nodes in multiple availablity zones and cloud providers.
+
+A very brief, non-exhaustive list of pro's and con's of the two
+approaches follows.  (In the interestes of full disclosure, the author
+prefers the Decoupled Hierarchical model for the reasons stated below).
+
+1. Failure isolation: The Decoupled Hierarchical approach provides
```

Comment:
nit: What about "Failure Likelihood"?  Comparing 3 clusters plus 1 FCP vs. 1 regional cluster, it would seem that for the former there are at least (1) more pieces to fail, (2) increased complexity (in terms of code and config) resulting in higher probability of bugs, and (3) more config so more room for operator error.

This is not a strict dominates-better-than, but a trade-off of different things: under one set of failure modes, one approach will suffer worse/more failures, and vice versa.